core/src/main/java/com/datastrato/gravitino/storage/relational/service/RoleMetaService.java

```diff
@@ -145,8 +205,138 @@ public int deleteRoleMetasByLegacyTimeLine(long legacyTimeLine, int limit) {
                 SessionUtils.doWithoutCommitAndFetchResult(
                     GroupRoleRelMapper.class,
                     mapper ->
-                        mapper.deleteGroupRoleRelMetasByLegacyTimeLine(legacyTimeLine, limit)));
+                        mapper.deleteGroupRoleRelMetasByLegacyTimeLine(legacyTimeLine, limit)),
+        () ->
+            securableObjectsCount[0] =
+                SessionUtils.doWithoutCommitAndFetchResult(
+                    SecurableObjectMapper.class,
+                    mapper ->
+                        mapper.deleteSecurableObjectsByLegacyTimeLine(legacyTimeLine, limit)));
+
+    return roleDeletedCount[0]
+        + userRoleRelDeletedCount[0]
+        + groupRoleRelDeletedCount[0]
+        + securableObjectsCount[0];
+  }
+
+  long getSecurableObjectEntityId(long metalakeId, String fullName, MetadataObject.Type type) {
```

Comment:
a) Why there's no access level modifier for this method?
b) This and below methods are generic to role, you should make them generic and put into a utility class, so that tag implementation can also leverage these methods.